WordPress/gutenberg
# Try: Update Tertiary Button appearance

## What?
Try some updates to the Tertiary Button appearance.

## Why?
There are a couple of issues I'm seeking to address:

* Tertiary Buttons currently have a unique padding value which looks a bit sloppy / unintentional when they appear next to other buttons.
* The hover effect features an outline, which overlaps a bit with the Secondary button style.
* The `active` state features a blue-on-grey appearance which looks a little muddy.

## How?
Taking inspiration from the updated Block Inserter, a solid background is applied instead of an outline on hover. Additionally the padding override is removed so that Tertiary buttons share padding with all other text buttons.

It's worth noting that the padding override was originally added to improve the optics when Tertiary buttons appear consecutively, e.g. here:

<img width="380" alt="Screenshot 2023-03-07 at 13 20 46" src="https://user-images.githubusercontent.com/846565/223434273-cb36793a-a94c-415a-9ef1-63944b3fa0a9.png">

But this feels like something that should be addressed at the 'organism' level rather than the component level. 

## Testing Instructions
* Open the Post or Site Editor
* Observe and interact with any Tertiary buttons

### Testing Instructions for Keyboard
* Focus a Tertiary button
* Ensure it is visually obvious that the button has focus

## Screenshots or screencast 

| Before | After |
| --- | --- |
| ![before](https://user-images.githubusercontent.com/846565/223435613-99492edf-04c7-4985-8446-02eed4350446.gif) | ![after](https://user-images.githubusercontent.com/846565/223435654-6043bad4-2167-42b5-a9a3-7ab4155f1cee.gif) |
